File: Data Structures & Algorithms/longest-common-prefix/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string}
     */
    longestCommonPrefix(strs) {
       let str = ""; 
       let w = strs[0]; 
       if (w.length === 0) return ""; 
       for (let i = 0 ; i < w.length ; i++){
            let char = w[i]; 
            for (let j = 0 ; j < strs.length ; j++){
                if (strs[j][i] != char){
                    return str ; 
                }
            }
            str += char ; 
       }
       return str; 
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        # first count the len 
        # the add teh len and then a # 
        # add teh string 

        encoded = ""

        for s in strs : 
            length = len(s)
            encoded  += str(length)
            encoded += str("#")
            encoded += s 
        return encoded 
        
            

    def decode(self, s: str) -> List[str]:
        results  =  []
        i = 0 
        while ( i < len(s)) : 
            j = i 
            while s[j] != "#" : 
                j += 1 
            
            length =  int(s[i : j])

            j += 1 
            word  = s[j : j + length]
            results.append(word)

            i = j + length  

        return results 
